### PR TITLE
gh-99606: Make code generated for an empty f-string identical to that of a normal empty string

### LINKED ISSUE #99606: Bytecode for empty f-string != bytecode for empty normal string
# Bug report

The bytecode for an empty f-string does not match the bytecode for an empty normal string.
```py
import dis
dis.dis('""')
dis.dis('f""')
```
Output:
```
0           0 RESUME                   0

1           2 LOAD_CONST               0 ('')
            4 RETURN_VALUE
0           0 RESUME                   0

1           2 BUILD_STRING             0
            4 RETURN_VALUE
```

# Environment

- CPython versions tested on:
3.11, 3.12.0a2

### Linked PRs
* gh-99608
* gh-112407

### PR DESCRIPTION
**Pull request by iritkatriel**
Fixes #99606.

* Issue: gh-99606

### CODE REVIEW

File: Lib/importlib/_bootstrap_external.py [hunk 1]
```diff
@@ -477,7 +478,7 @@ def _write_atomic(path, data, mode=0o666):
 # Whenever MAGIC_NUMBER is changed, the ranges in the magic_values array
 # in PC/launcher.c must also be updated.
 
-MAGIC_NUMBER = (3565).to_bytes(2, 'little') + b'\r\n'
+MAGIC_NUMBER = (3578).to_bytes(2, 'little') + b'\r\n'
```
Comment (iritkatriel) [inline Lib/importlib/_bootstrap_external.py]:
```suggestion
MAGIC_NUMBER = (3566).to_bytes(2, 'little') + b'\r\n'
```

**Review by ericvsmith — APPROVED** [review]
I'm not sure why the magic number needs to increase, because this is a backwards compatible change, but I suppose it doesn't really matter.

In any event, this looks good to me. Thanks!

File: Lib/importlib/_bootstrap_external.py [hunk 2]
```diff
@@ -477,7 +478,7 @@ def _write_atomic(path, data, mode=0o666):
 # Whenever MAGIC_NUMBER is changed, the ranges in the magic_values array
 # in PC/launcher.c must also be updated.
 
-MAGIC_NUMBER = (3565).to_bytes(2, 'little') + b'\r\n'
+MAGIC_NUMBER = (3566).to_bytes(2, 'little') + b'\r\n'
```
Comment (iritkatriel) [inline Lib/importlib/_bootstrap_external.py]:
```suggestion
MAGIC_NUMBER = (3565).to_bytes(2, 'little') + b'\r\n'
```

File: Lib/importlib/_bootstrap_external.py [hunk 3]
```diff
@@ -461,6 +461,7 @@ def _write_atomic(path, data, mode=0o666):
 #     Python 3.13a1 3563 (Add CALL_KW and remove KW_NAMES)
 #     Python 3.13a1 3564 (Removed oparg from YIELD_VALUE, changed oparg values of RESUME)
 #     Python 3.13a1 3565 (Oparg of YIELD_VALUE indicates whether it is in a yield-from)
+#     Python 3.13a1 3566 (Code for empty f-string matches empty normal string)
```
Comment (iritkatriel) [inline Lib/importlib/_bootstrap_external.py]:
```suggestion
```

**Pull request merged**